gossip: plumb contexts

RaduBerinde opened the pull request:
Plumbing contexts through gossip. Attaching an EventLog to the context for all
gossip logs, removing the eventLog.

Part of #7995.

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/8869)

<!-- Reviewable:end -->

RaduBerinde committed:
gossip: plumb contexts

Plumbing contexts through gossip. Attaching an EventLog to the context for all
gossip logs, removing the eventLog.

Part of #7995.

tschottdorf commented:
<img class="emoji" title=":lgtm_strong:" alt=":lgtm_strong:" align="absmiddle" src="https://reviewable.io/lgtm_strong.png" height="20" width="61"/> 

---

Reviewed 19 of 19 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---

*[gossip/gossip.go, line 894 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8869#-KQBnipgmZmKu91qlnMg:-KQBnipgmZmKu91qlnMh:budyasv) ([raw file](https://github.com/cockroachdb/cockroach/blob/d6534e5fa7fa65140e96a35599012edeccc7353f/gossip/gossip.go#L894)):*
> ```Go
> 					// Try to get another bootstrap address from the resolvers.
> 					if addr := g.getNextBootstrapAddress(); addr != nil {
> 						g.startClient(ctx, addr, g.mu.is.NodeID)
> ```

now this client would have the `bootstrap` tag which long outlasts when that client was created; better to use `g.ctx` here?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/8869#-:-KQBo8qW7bA7xSoU7R8l:b-uwkowd)*
<!-- Sent from Reviewable.io -->

RaduBerinde commented:
Review status: 16 of 19 files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---

*[gossip/gossip.go, line 894 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8869#-KQBnipgmZmKu91qlnMg:-KQBxJ2k0OKt96AgrQwT:bhrv71b) ([raw file](https://github.com/cockroachdb/cockroach/blob/d6534e5fa7fa65140e96a35599012edeccc7353f/gossip/gossip.go#L894)):*
<details><summary><i>Previously, tschottdorf (Tobias Schottdorf) wrote…</i></summary>
> now this client would have the `bootstrap` tag which long outlasts when that client was created; better to use `g.ctx` here?
</details>
Sure, in fact this call site is the only reason I added an arg to `startClient`. Simplified to just use `g.ctx` internally.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/8869)*
<!-- Sent from Reviewable.io -->

tschottdorf commented:
Reviewed 3 of 3 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/8869)*
<!-- Sent from Reviewable.io -->
